src/renderer/components/menu/menu.tsx

```diff
@@ -85,6 +86,10 @@ export class Menu extends React.Component<MenuProps, State> {
     return !!this.props.isOpen;
   }
 
+  shouldComponentUpdate(nextProps: MenuProps, nextState: State) {
```

Comment:
Do we really need this? The real problem is invoking `refreshPosition()` from `render()`, right? Why does that cause an infinite render loop? if it doesn't change state or props then `shouldComponentUpdate()` shouldn't be needed(?)

Reply:
```setState``` will always cause a rerender even if the state is the same (without this method), but the virtual DOM won't do any operations if the result of the render does not change. 

The underlying reason is calling ```refreshPosition``` from the ```render```, and that was added apparently because sometimes the content will change asynchronously. The whole problem stems from manually moving the DOM element in the ```refreshPosition``` so the actual position state is in the DOM but not in the component state. So in order to sync the state in the DOM we are calling the ```refreshPosition``` way too much.

Actually now I realised the better solution is to calculate the position on the render instead of calling setState and remove the ```refreshPosition``` method call. I'll try this approach. Thanks!

Reply:
I'd rather to use `mobx.observable` to handle state changes.
Let try to avoid old-funky `setState` as possible?

Reply:
@ixrock, glad you're back!